repository/src/main/java/org/alfresco/repo/domain/permissions/FixedAclUpdaterListener.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * #%L
+ * Alfresco Repository
+ * %%
+ * Copyright (C) 2005 - 2016 Alfresco Software Limited
+ * %%
+ * This file is part of the Alfresco software.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail.  Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * Alfresco is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+package org.alfresco.repo.domain.permissions;
+
+import org.alfresco.service.cmr.repository.NodeRef;
+
+/** Listener to receive callback events when permissions are asynchronously updated with the {@link FixedAclUpdater}. */
+public interface FixedAclUpdaterListener
+{
+    /** Callback method for when permissions have been updated by the FixedAclUpdater. */
+    void permissionsUpdated(NodeRef nodeRef);

```

Comment:
Could be better to highlight that this method is invoked only when the update is asynchronous? We can call it something like permissionsUpdatedAsynchronously. The comment is pretty clear but when I read the implementation on the other PR isn't so clear to me at first glance.